### DISCUSSION TITLE
Use 11ty to make many "micro-sites" but can still share content between them.

### DISCUSSION THREAD

**Discussion by dtlevin** [11ty/eleventy#2259] [Q&A]
We use 11ty to make "micro-sites" for clients. Even though each site is just for one client they all still share common content. An example would be that they all have certain case studies that we show on each site. Each site would have 4-5 out 10 that are there now. I would like to have all of these in one directory outside the "micro-sites" so they are all pulling from a common library. The same for images, CSS, Js, etc. I can't how to write a filter for these in the documentation. Any help or ideas would be appreciated.

**Comment by cfjedimaster** [comment]
A couple of things come to mind:

* What about a symbolic link?
* I don't know if you can, but you could possibly require(..) above the root and into the shared directory?
* As part of your build process, copy the assets to the micro site? That would cost file storage, but that's cheap. If you always copy on build (and maybe delete first), you'll have the latest every time you build. 
* Maybe put the asset on it's own site, and hook up an API that returns them so that your microsite can make a network request to copy them down?

**Reply by pdehaan** [reply]
I think there have been some examples in past issues where we faked it by setting different `--input` and `--output` directories via the [command line](https://www.11ty.dev/docs/usage/), and using shared _data/ and _includes/ folders.
Not sure if that works for you, but might give you some ideas.

**Reply by pdehaan** [reply]
Actually, I might be mis-remembering. I wasn't able to get a shared _includes/ and _layouts/ directory working when overriding `--input` on the CLI (even w/ absolute or ../_includes relative-style links). I think symbolic links might be the way to go.

It certainly isn't pretty, but here was my rough attempt using Eleventy 1.0 and hacking some global data using `.addGlobalData()` and the [Eleventy API](https://www.11ty.dev/docs/programmatic/):
https://github.com/pdehaan/11ty-multisite

**Comment by madsem** [comment]
The way I solved a similar case:

I created a github organization, to be able to have shared repository secrets.

Then a main repository, that holds all of the 11ty templates and code.

A content repository, that holds only the _data directory and directories with markdown content, shared assets etc. This is a template repo... So to create new sites i just use the "use this template" button in github to create a new content repository.

Wrote a Github workflow that clones the main repository every time a push is made into one of my content repositories. Remove the .git directory, then copy the content repository into the cloned main repository folder, and push that into a "deployment" branch of the content repository.

I use Netlify, so on push into deployment branch Netlify makes a new build...

Was a bit of work, but now it's all very smooth.
